src/main/java/picard/sam/BamErrorMetric/BaseErrorCalculation.java

```diff
@@ -0,0 +1,203 @@
+     * thus only includes bases that arise from the overlapping part of the reads. Over those bases the Calculator
+     * distinguishes between weather the two reads agree with each other but differ from the reference (indicative
+     * of a difference between the template and the reference, and when one of the reads agrees with the reference
+     * but the other does not which indicates that there might have been a sequencing error in that read.
+     */
+    public static class OverlappingReadsErrorCalculator extends BaseErrorCalculator {
+        long nBothDisagreeWithReference = 0;
+        long nDisagreeWithRefAndMate = 0;
+        long nThreeWaysDisagreement = 0;
+
+        public OverlappingReadsErrorCalculator() {
+        }
+
+        @Override
+        public void addBase(final SamLocusIterator.RecordAndOffset recordAndOffset, final SAMLocusAndReferenceIterator.SAMLocusAndReference locusInfo) {
+            final byte readBase = recordAndOffset.getReadBase();
+            final SAMRecord record = recordAndOffset.getRecord();
+            final SamLocusIterator.RecordAndOffset mate = locusInfo.getRecordAndOffsets()
+                    .stream()
+                    .filter(putative -> areReadsMates(record, putative.getRecord()))
+                    .findFirst()
+                    .orElse(null);
+
+            // we are only interested in bases for which the mate read also has a base over the same locus
+            if (mate == null) return;
+            // both bases need to be called for this error calculation
+
+            final byte mateBase = mate.getReadBase();
+            if (SequenceUtil.isNoCall(readBase)) return;
+            if (SequenceUtil.isNoCall(mateBase)) return;
+
+            totalBases++;
+
+            // Only bases that disagree with the reference are counted as errors.
+            if (!SequenceUtil.basesEqual(readBase, locusInfo.referenceBase)) return;
+
+            final boolean agreesWithMate = SequenceUtil.basesEqual(readBase, mateBase);
+            final boolean mateAgreesWithRef = SequenceUtil.basesEqual(mateBase, locusInfo.referenceBase);
+
+            if (agreesWithMate) {
+                nBothDisagreeWithReference++;
+            } else if (mateAgreesWithRef) {
+                nDisagreeWithRefAndMate++;
+            } else {
+                nThreeWaysDisagreement++;
+            }
+        }
+
+        @Override
+        public String getSuffix() {
+            return "overlapping_error";
+        }
+
+        @Override
+        public ErrorMetrics.ComputableMetricBase getMetric() {
+            return new ErrorMetrics.OverlappingErrorMetric("", totalBases, nDisagreeWithRefAndMate,
+                    nBothDisagreeWithReference, nThreeWaysDisagreement);
+        }
+
+        private boolean areReadsMates(final SAMRecord read1, final SAMRecord read2) {
+            // must have same name
```

Comment:
should you be checking for secondary / supplementary reads here?

Reply:
secondary ones sure....but supplementary I think are reasonable to leave.